packages/amplify-provider-awscloudformation/src/aws-utils/S3Service.ts

```diff
@@ -27,33 +20,25 @@ export class S3Service implements IS3Service {
       const response = await this.s3.listBuckets().promise();
 
       if (response.Buckets) {
-        this.cachedBucketList.push(...response.Buckets!);
+        this.cachedBucketList.push(...response.Buckets);
       }
     }
 
     return this.cachedBucketList;
   }
 
-  public async bucketExists(bucketName: string): Promise<boolean> {
+  public async bucketExists(bucketName: string, s3?: S3): Promise<boolean> {
```

Comment:
The point of `S3Service` abstraction is to hide `S3` . We should not pass `S3` as parameter at public API.

Two options:
1. create private `bucketExists(bucketName: string, s3?: S3)` and call into it from public version.
2. when we re-try call into `createS3Service` with overriden region and use it to retry exist call.